satpy/dataset/metadata.py

```diff
@@ -141,6 +149,27 @@ def _all_arrays_equal(arrays):
         return _pairwise_all(nan_allclose, arrays)
 
 
+def _all_dicts_equal(dicts):
+    return _pairwise_all(_dict_equal, dicts)
+
+
+def _dict_equal(d1, d2):
+    """Check that two dictionaries are equal.
+
+    Nested dictionaries are walked recursively. Array type values are compared
+    approximately, other types are checked for identity.
+    """
+    if not (isinstance(d1, dict) and isinstance(d2, dict)):
+        return False
+    if d1.keys() != d2.keys():
+        return False
+    for key in d1.keys():
+        if isinstance(d1[key], dict) and isinstance(d2[key], dict):
+            return _dict_equal(d1[key], d2[key])
```

Comment:
Don't these two returns need to only return if `False`? Otherwise we aren't checking every key.